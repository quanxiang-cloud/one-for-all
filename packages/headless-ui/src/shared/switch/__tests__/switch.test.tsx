import React from 'react';
import { render, fireEvent, waitFor, act } from '@testing-library/react';
import Switch from '../index';

describe('Switch', () => {
  it('switch without anything', () => {
    const wrapper = render(<Switch />);
    expect(wrapper.asFragment()).toMatchSnapshot();
  });

  it('component could be render and unmounted without errors', () => {
    expect(() => {
      const wrapper = render(<Switch />);
      wrapper.unmount();
    }).not.toThrow();
  });

  it('switch with checked and onText', () => {
    const wrapper = render(<Switch checked onText="开" />);
    expect(wrapper.asFragment()).toMatchSnapshot();
    expect(wrapper.container.querySelector('.ofa-switch-checked')).toBeInTheDocument();
  });

  it('switch with false checked and offText', () => {
    const wrapper = render(<Switch offText="关" />);
    expect(wrapper.asFragment()).toMatchSnapshot();
  });

  it('should get input element ref', () => {
    const ref = React.createRef<HTMLLabelElement>();
    render(<Switch ref={ref} />);
    expect(ref.current instanceof HTMLLabelElement).toBeTruthy();
  });

  it('should be change checked while click', () => {
    const wrapper = render(<Switch />);

    act(() => {
      fireEvent.click(wrapper.container);
    });
    waitFor(() => expect(wrapper.container.querySelector('.ofa-switch-checked')).toBeInTheDocument());
    act(() => {
      fireEvent.click(wrapper.container);
    });
    waitFor(() => expect(wrapper.container.querySelector('.ofa-switch-checked')).not.toBeInTheDocument());
  });
});
