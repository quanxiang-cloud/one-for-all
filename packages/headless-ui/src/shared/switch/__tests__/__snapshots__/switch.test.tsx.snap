// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Switch switch with checked and onText 1`] = `
<DocumentFragment>
  <label
    class="ofa-switch ofa-switch-checked"
  >
    <span
      class="ofa-switch-text ofa-switch-text-checked"
    >
      开
    </span>
  </label>
</DocumentFragment>
`;

exports[`Switch switch with false checked and offText 1`] = `
<DocumentFragment>
  <label
    class="ofa-switch"
  >
    <span
      class="ofa-switch-text"
    >
      关
    </span>
  </label>
</DocumentFragment>
`;

exports[`Switch switch without anything 1`] = `
<DocumentFragment>
  <label
    class="ofa-switch"
  >
    <span
      class="ofa-switch-text"
    />
  </label>
</DocumentFragment>
`;
